tests/ui/type-alias-impl-trait/issue-53398-cyclic-types.rs

```diff
@@ -1,9 +1,10 @@
 #![feature(type_alias_impl_trait)]
 
+// check-pass
+
 type Foo = impl Fn() -> Foo;
```

Comment:
how is that fixed? Though I guess this should work, so it's fine

Reply:
it's fixed by not normalizing eagerly `Foo`. So normalize the weak alias to the opaque type, and if we get the `item_bounds`, we get an unnormalized `Foo`.

Previously, when registering a hidden type, we used to instantiate the opaque type, which then contained itself, which we tried to instantiate, too, and thus cycled.